pkg/k8s/watchers/node.go

```diff
@@ -127,25 +119,26 @@ func (k *K8sWatcher) GetK8sNode(_ context.Context, nodeName string) (*v1.Node, e
 // ciliumNodeUpdater implements the subscriber.Node interface and is used
 // to keep CiliumNode objects in sync with the node ones.
 type ciliumNodeUpdater struct {
-	k8sWatcher         *K8sWatcher
-	kvStoreNodeUpdater KVStoreNodeUpdater
+	kvStoreNodeUpdater NodeUpdate
 }
 
-func NewCiliumNodeUpdater(k8sWatcher *K8sWatcher, kvStoreNodeUpdater KVStoreNodeUpdater) *ciliumNodeUpdater {
+func NewCiliumNodeUpdater(kvStoreNodeUpdater NodeUpdate) *ciliumNodeUpdater {
 	return &ciliumNodeUpdater{
-		k8sWatcher:         k8sWatcher,
 		kvStoreNodeUpdater: kvStoreNodeUpdater,
 	}
 }
 
 func (u *ciliumNodeUpdater) OnAddNode(newNode *v1.Node, swg *lock.StoppableWaitGroup) error {
-	u.updateCiliumNode(u.kvStoreNodeUpdater, newNode)
-
+	// We don't need to run OnAddNode because Cilium will fetch the state from
+	// k8s upon initialization and will populate the KVStore [1] node with this
+	// information or create a Cilium Node CR [2].
+	// [1] https: // github.com/cilium/cilium/blob/2bea69a54a00f10bec093347900cc66395269154/daemon/cmd/daemon.go#L1102
```

Comment:
Thanks for adding these links!

Minor nit:
```suggestion
	// [1] https://github.com/cilium/cilium/blob/2bea69a54a00f10bec093347900cc66395269154/daemon/cmd/daemon.go#L1102
```